dungeon/src/contrib/hud/UIUtils.java

```diff
@@ -64,30 +71,89 @@ public static void center(final Actor actor) {
+   * Wrap texts to a maximum line length.
+   *
+   * <p>This function can be used to soft-wrap texts to a maximum of {@link UIUtils#MAX_ROW_LENGTH}
+   * characters per line. The text is sanitised before wrapping by replacing multiple consecutive
+   * whitespace characters (including line breaks) with single spaces. The text is then wrapped
+   * according to the following algorithm: The last space before the maximum line length of {@link
+   * UIUtils#MAX_ROW_LENGTH} characters is replaced by a line break. Any words exceeding the maximum
+   * line length are cut off and wrapped at the position of the maximum line length.
+   *
+   * @param text text to be soft-wrapped at {@link UIUtils#MAX_ROW_LENGTH} characters per line
+   * @return the reformatted text where all lines have been soft-wrapped to at maximum {@link
+   *     UIUtils#MAX_ROW_LENGTH} characters per line
+   */
+  public static String formatString(final String text) {
+    return formatString(text, MAX_ROW_LENGTH);
+  }
+
+  /**
+   * Wrap texts to a maximum line length of {@code maxLen} characters.
+   *
+   * <p>This function can be used to soft-wrap texts to a maximum of {@code maxLen} characters per
+   * line. The text is sanitised before wrapping by replacing multiple consecutive whitespace
+   * characters (including line breaks) with single spaces. The text is then wrapped according to
+   * the following algorithm: The last space before the maximum line length of {@code maxLen}
+   * characters is replaced by a line break. Any words exceeding the maximum line length are cut off
+   * and wrapped at the position of the maximum line length.
    *
-   * @param string String which should be reformatted.
+   * @param text text to be soft-wrapped at {@code maxLen} characters per line
+   * @param maxLen maximum number of characters per line
+   * @return the reformatted text where all lines have been soft-wrapped to at maximum {@code
+   *     maxLen} characters per line
    */
-  public static String formatString(final String string) {
-    StringBuilder formattedMsg = new StringBuilder();
-    String[] lines = string.split(System.lineSeparator());
-
-    for (String line : lines) {
-      String[] words = line.split(" ");
-      int sumLength = 0;
-
-      for (String word : words) {
-        sumLength += word.length();
-        formattedMsg.append(word);
-        formattedMsg.append(" ");
-
-        if (sumLength > MAX_ROW_LENGTH) {
-          formattedMsg.append(System.lineSeparator());
-          sumLength = 0;
-        }
+  public static String formatString(final String text, int maxLen) {
+    if (text == null) {
+      return null;
+    }
+
+    final boolean softWordWrap = true;
+
+    final StringBuilder sb = new StringBuilder();
+    final String[] words = text.trim().replaceAll("\\s+", " ").split(" ");
+
+    int wordIndex = 0;
+    int lineIndex = 0;
+
+    while (wordIndex < words.length) {
+      final String word = words[wordIndex++];
```

Comment:
wenn die deklaration der hilfsvariable eine zeile höher wäre (dann freilich ohne `final`), dann müsste das nicht von der laufzeitumgebung in jedem durchlauf immer wieder neu angelegt werden ...

Reply:
das ist doch nur eine refrenz und soll eigentlich die lesbarkeit erhöhen.

Reply:
hmmm, ja. aber die frage ist, ob die in jeder schleife neu angelegt werden muss?

Reply:
darf (wegen der lesbarkeit ...) innerhalb der schleife bleiben.